packages/account/src/Components/network-status-toast-popup/__tests__/network-status-toast-popup.spec.tsx

```diff
@@ -0,0 +1,115 @@
+        popup_root_el.setAttribute('id', 'popup_root');
+        document.body.appendChild(popup_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(popup_root_el);
+    });
+
+    const mock_store = mockStore({
+        common: { network_status: { class: offline.toLowerCase(), tooltip: offline } },
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const wrapper = ({ children }: { children: JSX.Element }) => (
+        <StoreProvider store={mock_store}>{children}</StoreProvider>
+    );
+
+    it('should render NetworkStatusToastPopup when offline', () => {
+        render(<NetworkStatusToastPopup />, { wrapper });
+
+        expect(screen.queryByText(connecting)).not.toBeInTheDocument();
+        expect(screen.queryByText(online)).not.toBeInTheDocument();
+        expect(screen.getByText(offline)).toBeInTheDocument();
+    });
+
+    it('should render NetworkStatusToastPopup when trying to connect', async () => {
+        jest.useFakeTimers();
+
+        mock_store.common.network_status.class = connecting.toLowerCase();
+        mock_store.common.network_status.tooltip = connecting;
+
+        const { rerender } = render(
+            <StoreProvider store={mock_store}>
+                <NetworkStatusToastPopup />
+            </StoreProvider>
+        );
+
+        expect(screen.queryByText(offline)).not.toBeInTheDocument();
+        expect(screen.queryByText(online)).not.toBeInTheDocument();
+        expect(screen.getByText(connecting)).toBeInTheDocument();
+
+        const mock_store_updated = mockStore({
+            common: { network_status: { class: online.toLowerCase(), tooltip: online } },
+        });
+
+        rerender(
+            <StoreProvider store={mock_store_updated}>
+                <NetworkStatusToastPopup />
+            </StoreProvider>
+        );
+
+        expect(screen.queryByText(offline)).not.toBeInTheDocument();
+        expect(screen.queryByText(connecting)).not.toBeInTheDocument();
+        expect(screen.getByText(online)).toBeInTheDocument();
+
+        act(() => {
+            jest.advanceTimersByTime(1500);
+        });
+
+        expect(screen.queryByText(offline)).not.toBeInTheDocument();
+        expect(screen.queryByText(connecting)).not.toBeInTheDocument();
+        expect(screen.queryByText(online)).not.toBeInTheDocument();
+
+        jest.useRealTimers();
+    });
+
+    it('should not render NetworkStatusToastPopup when no message', () => {
+        mock_store.common.network_status.class = offline.toLowerCase();
+        mock_store.common.network_status.tooltip = '';
+
+        render(<NetworkStatusToastPopup />, { wrapper });
+
+        expect(screen.queryByText(offline)).not.toBeInTheDocument();
+        expect(screen.queryByText(connecting)).not.toBeInTheDocument();
+        expect(screen.queryByText(online)).not.toBeInTheDocument();
+    });
+
+    it('should not render NetworkStatusToastPopup for desktop', () => {
```

Comment:
why do we not show NetworkStatusToastPopup for desktop?

Reply:
this is the current behaviour on production,
I asked this question, and it was decided to proceed with current behaviour for now. There is no design for desktop